api/resource.rb

```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
     module Properties
       include Api::Object::Named::Properties
 
+      # [Required] A description of the resource that's surfaced in provider
```

Comment:
The validate methods show the required/optional. Do we still think we need them in descriptions?

Reply:
Yes- the moment you start reading the validate methods you see hard to understand blocks like this which distract from that kind of information.

```ruby
      check :nested_query, type: Api::Resource::NestedQuery
      if @nested_query&.is_list_of_ids && @identity&.length != 1
        raise ':is_list_of_ids = true implies resource`\
              `has exactly one :identity property"'
      end
```